torch/_subclasses/fake_tensor.py

```diff
@@ -322,12 +391,15 @@ def check_non_fake_tensor(x):
                 if run_impl_check(func):
                     return op_impl(self, func, *args, **kwargs)
 
+            self.in_kernel_invocation = True
```

Comment:
Naïvely I would've expected these to pair up with the no dispatch call

Reply:
do you mean put inside the `no_dispatch` or to use `_DisableTorchDispatch` instead of `self.in_kernel_invocation` ? I haven't thought too hard about it... maybe would be a good cleanup, although I still need to do a little more thinking/understanding of composability